### PR TITLE
Remove ChunkedToXContentHelper.array method, swap for Builder

### PR DESCRIPTION
**Pull request by thecoop**

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticmachine update branch

**Comment by thecoop** [comment]
@elasticmachine update branch

**Review by ldematte — APPROVED** [review]
Looks good, but I'm not sure I would change the Iterator in the *Decision.java classes, it is one of those cases where it looks more natural.

File: server/src/main/java/org/elasticsearch/cluster/routing/allocation/AbstractAllocationDecision.java [hunk 1]
```diff
@@ -135,12 +133,12 @@ public static List<NodeAllocationResult> sortNodeDecisions(List<NodeAllocationRe
      * Generates X-Content chunks for the node-level decisions, creating the outer "node_allocation_decisions" object
      * in which they are serialized.
      */
-    public static Iterator<ToXContent> nodeDecisionsToXContentChunked(List<NodeAllocationResult> nodeDecisions) {
+    public static Consumer<ChunkedToXContentBuilder> nodeDecisionsToXContentChunked(List<NodeAllocationResult> nodeDecisions) {
```
[the remark this answers is not in the document]
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/cluster/routing/allocation/AbstractAllocationDecision.java:136]: Yeah, good idea

**Pull request merged**